fusilli-plugin/docs/TheRockRFC.md

```diff
@@ -0,0 +1,34 @@
+
+1) IREE has been built from the ground-up as a fusion compiler. The kinds of fusions that libraries like hipDNN are expected to provide are supported out-of-the box in IREE. 
+2) Being a compiler approach, IREE as a kernel provided functions as a JIT-backend. For cases where IREE generated code is faster, IREE would JIT-compile the kernels without having to pre-build kernels and ship those with hipDNN.
+
+This RFC is to propose adding a path to using IREE as a kernel provider to hipDNN within TheRock. There are three components to reason about
+
+1. The hipDNN backend plugin to Fusilli. Currently it lives [here](https://github.com/nod-ai/shark-ai/tree/main/fusilli-plugin)
+2. Fusilli. Currently lives [here](https://github.com/nod-ai/shark-ai/tree/main/sharkfuser)
+3. IREE, which lives in its own Github Org and is a Linux Foundation Project. It currently lives [here](https://github.com/iree-org/iree)
+
+## Workplan
+
+### Immediate next steps
+
+The immediate workplan is to move just the hipDNN backend plugin to Fusilli (so component 1 above) into TheRock. The plugin will be built conditionally (not on by default) and will pull in Fusilli as an external dependency. Nested dependencies of Fusilli are (TODO: Adjust/clarify the following).
+
+1. IREE runtime sources are built into Fusilli
+2. IREE compile using the command-line interface, i.e. the `iree-compile` binary needs to be available (typically through pip-install of the IREE compiler package)
+
+### Medium term/Long term requirements.
+
+While the initial integration will just focus on pulling in the hipDNN plugin to IREE into monorepo, long term the expectation is that Fusilli and IREE will not be pulled in as external binary/library dependencies. Some question that need to be answered for those are
+1. Where does Fusilli live? Fusilli is a C++ API around IREE and such is tightlt coupled with IREE. A natural home for Fusilli is within the same repo/github organization as IREE itself.
+2. The expectation is that Fusilli will stop using `iree-compile` as a binary, but rather use the C-API of the IREE compiler to JIT compile the (fused) kernel computation. This would require significant changes to current IREE workflow. Apart from resolving where the IREE project lives, i.e. if it should move into the monorepo as well, another challenge to solve there is which LLVM version should IREE use. IREE currently tracks top-of-main of LLVM pretty closely. This would need to change to use either the LLVM version within monorepo or a release version of LLVM/MLIR.
```

Comment:
Just a thought - it may be unnecessary / impractical to expect IREE to move into the monorepo, given it's in a neutral place right now from a governance standpoint. So we'll have to find alternate means to integrate here.